modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/redshift/Main.scala

```diff
@@ -0,0 +1,142 @@
+  FoldersCopy,
+  FoldersMinusManifest
+}
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+import fs2.Stream
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import com.snowplowanalytics.snowplow.rdbloader.core.config.StorageTarget.Redshift
+import com.snowplowanalytics.snowplow.rdbloader.core.algebras.FolderMonitoring.TempTableOps
+
+object Main extends IOApp {
+
+  implicit private val LoggerName =
+    Logging.LoggerName(getClass.getSimpleName.stripSuffix("$"))
+
+  def run(argv: List[String]): IO[ExitCode] =
+    CliConfig.parse(argv) match {
+      case Valid(cli) =>
+        Environment.initialize[IO](cli).use { env =>
+          import env._
+
+          Logging[IO].info(s"RDB Loader ${BuildInfo.version} has started. Listening to ${cli.config.messageQueue}") *>
+            process[IO](cli, control).compile.drain.as(ExitCode.Success).handleErrorWith(handleFailure[IO])
+        }
+      case Invalid(errors) =>
+        val logger = Slf4jLogger.getLogger[IO]
+        logger.error("Configuration error") *>
+          errors.traverse_(message => logger.error(message)).as(ExitCode(2))
+    }
+
+  /**
+    * Main application workflow, responsible for discovering new data via message queue
+    * and processing this data with loaders
+    *
+    * @param cli whole app configuration
+    * @param control various stateful controllers
+    * @return endless stream waiting for messages
+    */
+  def process[F[_]: Concurrent: AWS: Iglu: Cache: Logging: Timer: Monitoring: JDBC](
+    cli: CliConfig,
+    control: Environment.Control[F]
+  ): Stream[F, Unit] = {
+    // Parse all configuration necessary for monitoring of corrupted and unloaded folders.
+    // If some configurations are not provided - just print a warning.
+    val folderMonitoring: Stream[F, Unit] = {
+      (cli.config.storage, cli.config.monitoring.folders) match {
+        case (r: Redshift, Some(folders)) =>
+          val create   = CreateAlertingTempTable
+          val drop     = DropAlertingTempTable
+          val copy     = FoldersCopy(folders.shredderOutput, r.roleArn)
+          val minus    = FoldersMinusManifest(r.schema)
+          val tableOps = TempTableOps(create, drop, copy, minus)
+
+          FolderMonitoring.run[F](
+            folders,
+            tableOps
+          )
+
+        case (_: Redshift, _) =>
+          Stream.eval[F, Unit](
+            Logging[F].info("Configuration for monitoring.folders hasn't been provided - monitoring is disabled")
+          )
+
+        case _ => Stream.eval[F, Unit](Logging[F].info("Configured storage target is not Redshift."))
+      }
+    }
+
+    val addLoadTstamp = Migration.addColumn[F](
+      cli.config.storage.schema,
+      "events",
+      Common.toAddColumn(AtomicDefaultColumns.loadTstamp, Some("GETDATE()"))
+    )
+
+    val manifest = Manifest.make
+    val load     = Load.make
+
+    Stream.eval_(manifest.initialize[F](cli.config.storage) *> addLoadTstamp.rethrowT) ++
```

Comment:
With the manifest and load being turned to algebras. This could be moved into core, with signature like run(config: StorageTarget)F[_]: Manifest: Load].